packages/blaze/template.js

```diff
@@ -313,6 +315,78 @@ Blaze.TemplateInstance.prototype.autorun = function (f) {
 };
 
 /**
+ * @summary A version of [Meteor.subscribe](#meteor_subscribe) that is stopped
+ * when the template is destroyed.
+ * @return {SubscriptionHandle} The subscription handle to the newly made
+ * subscription. Call `handle.stop()` to manually stop the subscription, or
+ * `handle.ready()` to find out if this particular subscription has loaded all
+ * of its inital data.
+ * @locus Client
+ * @param {String} name Name of the subscription.  Matches the name of the
+ * server's `publish()` call.
+ * @param {Any} [arg1,arg2...] Optional arguments passed to publisher function
+ * on server.
+ * @param {Function|Object} [callbacks] Optional. May include `onError` and
+ * `onReady` callbacks. If a function is passed instead of an object, it is
+ * interpreted as an `onReady` callback.
+ */
+Blaze.TemplateInstance.prototype.subscribe = function (/* arguments */) {
+  var subHandles = this._subscriptionHandles;
+  var args = _.toArray(arguments);
+
+  // Duplicate logic from Meteor.subscribe
+  var callbacks = {};
+  if (args.length) {
+    var lastParam = _.last(args);
+    if (_.isFunction(lastParam)) {
+      callbacks.onReady = args.pop();
+    } else if (lastParam &&
+      // XXX COMPAT WITH 1.0.3.1 onError used to exist, but now we use
+      // onStopped with an error callback instead.
+      _.any([lastParam.onReady, lastParam.onError, lastParam.onStopped],
+        _.isFunction)) {
+      callbacks = args.pop();
+    }
+  }
+
+  var oldStopped = callbacks.onStopped;
+  callbacks.onStopped = function (error) {
+    // When the subscription is stopped, remove it from the list of tracked
+    // subscriptions to avoid this list growing without bound
+    subHandles.set(_.without(subHandles, subHandle));
+
+    if (oldStopped) {
+      oldStopped(error);
+    }
+  };
+
+  args.push(callbacks);
+
+  var subHandle = this.view.subscribe.apply(this.view, args);
+
+  // If we are inside an autorun, the code inside would cause an infinite loop
+  Tracker.nonreactive(function () {
+    // If we get the same subhandle twice because we are in an autorun, make sure
+    // to only save it once
+    subHandles.set(_.union(subHandles.get(), [subHandle]));
+  });
+
+  return subHandle;
+};
+
+/**
+ * @summary A reactive function that returns true when all of the subscriptions
+ * called with [this.subscribe](#TemplateInstance-subscribe) are ready.
+ * @return {Boolean} True if all subscriptions on this template instance are
+ * ready.
+ */
+Blaze.TemplateInstance.prototype.subscriptionsReady = function () {
```

Comment:
It seems like this will often trigger deps changed without a value change.  eg, on unsub, it will stay true.

 what's the function that you can use to make that not happen?